internal/driver/glfw/window.go

```diff
@@ -658,17 +660,19 @@ func (w *window) mouseClicked(viewport *glfw.Window, btn glfw.MouseButton, actio
 		w.mouseLastClick = co
 	}
 
+	_, tap := co.(fyne.Tappable)
+	_, altTap := co.(fyne.SecondaryTappable)
 	// Prevent Tapped from triggering if DoubleTapped has been sent
-	if wid, ok := co.(fyne.Tappable); ok && doubleTapped == false {
+	if (tap || altTap) && doubleTapped == false {
 		if action == glfw.Press {
-			w.mousePressed = wid
+			w.mousePressed = co
 		} else if action == glfw.Release {
-			if wid == w.mousePressed {
+			if co == w.mousePressed {
 				switch button {
 				case desktop.RightMouseButton:
-					w.queueEvent(func() { wid.TappedSecondary(ev) })
+					w.queueEvent(func() { co.(fyne.SecondaryTappable).TappedSecondary(ev) })
 				default:
-					w.queueEvent(func() { wid.Tapped(ev) })
+					w.queueEvent(func() { co.(fyne.Tappable).Tapped(ev) })
```

Comment:
You should better add some test cases :).
This will panic if the obj search finds a `SecondaryTappable` but the mouse event was a primary click and vice versa.

Reply:
I'm actually going to defend this - the two objects will not be equal so 5 lines above will be false and this code will not execute...

I certainly could add tests to show the code doesn't run if that helps ;)

Reply:
Please write a test. I still think it will fail if you right-click on a left-click-only object and if you left-click on a right-click-only object.